src/spikeinterface/core/tests/test_sparsity.py

```diff
@@ -212,6 +240,9 @@ def test_compute_sparsity():
     sparsity = compute_sparsity(sorting_analyzer, method="best_channels", num_channels=2, peak_sign="neg")
     sparsity = compute_sparsity(sorting_analyzer, method="radius", radius_um=50.0, peak_sign="neg")
     sparsity = compute_sparsity(sorting_analyzer, method="snr", threshold=5, peak_sign="neg")
+    sparsity = compute_sparsity(
```

Comment:
Usually during a review I would run the code under the different arguments to check for any regressions introduced by the refactorings (it looks good to me reading the code, but I guess you never know!). However there are a lot of different options so it might be easier just to extend these tests. It might be worth now or in a follow up PR extending these tests to check that, given a sorting analyzer with some set of fixed templates, the results of all these methods are as expected (i.e. the machinery around the computation is properly propagating the kwargs etc).